### PR TITLE
Anikait/Sophia - Add Student Name to Ride Entity, fix ride table bug, add ride to navbar

### PR DESCRIPTION
**Pull request by strongani**
In this PR, we added the `student` attribute to the Ride entity, and the Ride controller get, so that the Student name could be seen in the RideTable on the frontend. Also fixed a bug with the path of the Create Ride button. 

This PR also inclused everything that was in PR #58 that updates the ride table to display all the backend information + tests + [storybook](https://ucsb-cs156-s23.github.io/proj-gauchoride-s23-5pm-4/prs/60/storybook/?path=/story/components-nav-appnavbar--no-role). We also added routes and the navbar for ride request. This finishes all frontend + backend for #7.

The only difference from this PR from #58 is the Ride entity + controller and tests, a change in the AppNavbar tests to fix mutation testing, and a fix in the RideRequestIndexPage with the create ride request button route.

### CODE REVIEW

**Review by SophiaWeiler — DISMISSED** [review]
lgtm

**Review by pconrad — APPROVED** [review]
LGTM--will test on staff qa

File: frontend/src/App.js [hunk 1]
```diff
@@ -24,6 +29,15 @@ function App() {
...
           hasRole(currentUser, "ROLE_ADMIN") && <Route exact path="/admin/users" element={<AdminUsersPage />} />
         }
+        {
+          (hasRole(currentUser, "ROLE_ADMIN") || hasRole(currentUser, "ROLE_DRIVER") || hasRole(currentUser, "ROLE_RIDER") )&& <Route exact path="/ride/" element={<RideRequestIndexPage />} />
+        }
+        {
+          (hasRole(currentUser, "ROLE_RIDER") || hasRole(currentUser, "ROLE_ADMIN"))  && <Route exact path="/ride/create" element={<RideRequestCreatePage />} />
+        }
+        {
+          (hasRole(currentUser, "ROLE_ADMIN")  || hasRole(currentUser, "ROLE_RIDER") )&& <Route exact path="/ride/edit:id" element={<RideRequestEditPage />} />
```
Comment (pconrad) [inline frontend/src/App.js]: I think this change will fix the bug where the edit button doesn't work:

```suggestion
          (hasRole(currentUser, "ROLE_ADMIN")  || hasRole(currentUser, "ROLE_RIDER") )&& <Route exact path="/ride/edit/:id" element={<RideRequestEditPage />} />
```

**Review by pconrad — CHANGES_REQUESTED** [review]
Currently, the edit button from the RideRequest page leads to a blank page, because the url

`https://proj-gauchoride-staff-qa.dokku-04.cs.ucsb.edu/ride/edit/2`

has no mapping in App.js. But if you manually edit the url to this:

`https://proj-gauchoride-staff-qa.dokku-04.cs.ucsb.edu/ride/edit2`

then it works.

So I think you need a one character change to fix this.  This is type of error that you find with end2end testing (which we don't have in our stack yet.)  In the meantime, it only shows up with manual testing.

**Pull request merged**